components/builder-jobsrv/src/data_store.rs

```diff
@@ -554,16 +552,11 @@ fn row_to_job(row: &postgres::rows::Row) -> Result<jobsrv::Job> {
     job.set_id(id as u64);
     let owner_id: i64 = row.get("owner_id");
     job.set_owner_id(owner_id as u64);
+
     let js: String = row.get("job_state");
-    let job_state = match &js[..] {
-        "Dispatched" => jobsrv::JobState::Dispatched,
-        "Pending" => jobsrv::JobState::Pending,
-        "Processing" => jobsrv::JobState::Processing,
-        "Complete" => jobsrv::JobState::Complete,
-        "Rejected" => jobsrv::JobState::Rejected,
-        "Failed" => jobsrv::JobState::Failed,
-        _ => return Err(Error::UnknownJobState),
-    };
+    let job_state = jobsrv::JobState::from_str(&js).map_err(
```

Comment:
Minor nit pick, but it seems like it's more idiomatic to use `parse` rather than calling `from_str` directly.